scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1255,7 +1155,16 @@ private void processDecryptedHandshakeMessage(final HandshakeMessage handshakeMe
 			final Connection connection) throws HandshakeException {
 		switch (handshakeMessage.getMessageType()) {
 		case CLIENT_HELLO:
-			processClientHello((ClientHello) handshakeMessage, record, connection);
+			// We do not support re-negotiation as recommended in :
+			// https://tools.ietf.org/html/rfc7925#section-17
+			if (record.getEpoch() > 0) {
+				DTLSSession session = connection.getEstablishedSession();
+				send(new AlertMessage(AlertLevel.WARNING, AlertDescription.NO_RENEGOTIATION, record.getPeerAddress()),
+						session);
+			} else {
+				LOGGER.error("Unexpected CLIENT_HELLO {}", record.getPeerAddress(),
+						new IllegalArgumentException("Unexpected CLIENT_HELLO"));
+			}
```

Comment:
I'm asking myself if this will avoid new connection (same peer address, fresh client hello) but we should have unit test for that ?

Reply:
All new CLIENT_HELLO are routed through processNewClientHello and not longer processHandshakeRecord. That's in receiveNextDatagramFromNetwork.

The new CLIENT_HELLOs don't require serialized execution until isClientInControlOfSourceIpAddress .